topo/node/cptx/cptx_test.go

```diff
@@ -139,6 +140,44 @@ func TestConfigPush(t *testing.T) {
 	}
 }
 
+func TestCustomPrivilegeLevel(t *testing.T) {
+	tests := []struct {
+		desc        string
+		hostname    string
+		shouldMatch bool
+	}{
+		{
+			desc:        "basic case",
+			hostname:    "testexample",
+			shouldMatch: true,
+		},
+		{
+			desc:        "hostname with '.'",
+			hostname:    "test.example",
+			shouldMatch: true,
+		},
+	}
+
+	privilegePromptMap := map[string]string{
+		"exec":          "root@%s>",
+		"configuration": "root@%s#",
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			for privilege, prompt := range privilegePromptMap {
+				match, err := regexp.Match(privLevels[privilege].Pattern, []byte(fmt.Sprintf(prompt, tt.hostname)))
```

Comment:
Safer to lookup the dict first in case key DNE:

```
fullPrompt := fmt.Sprintf(prompt, tt.hostname)
privLevel, ok := privLevels[privilege]
if !ok {
  t.Fatalf(...)
}
match, err := regexp.MatchString(privLevel.Pattern, fullPrompt)
if err != nil {
  t.Fatalf("regexp.MatchString() failed: %v", err)
}
if match != tt.shouldMatch {
  t.Fatal("regexp.MatchString() unexpected result (got %v, want %v) for prompt %q and privilege level %q", match, tt.shouldMatch, fullPrompt, privLevel)
}
```

Also note a few more patterns:
- error message should contain the func that failed
- got before want
- error message is clear exactly what case failed to the user (without even needing to lookup the test case by description name)